GoogleCloudPlatform/spring-cloud-gcp
# fix: add runtime hints for logging

~~Need rebase after setups in #1931 are merged in.~~

This pr adds:
- runtime hints for springframework aot to allow Logging module work with Native build.
- a relevant unit test
- runtime hints needed for sample integration test. (this is test setup specific and is not needed to run sample application).

### CI for native is setup to run on sample integration tests that:
- has `it.xx` property added to [systemPropertyVariables](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/f11fde736c783c4334ae50cb31e8ccf879738daa/spring-cloud-gcp-samples/pom.xml#L133) 
- module added to [native-sample-config](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/f11fde736c783c4334ae50cb31e8ccf879738daa/spring-cloud-gcp-samples/pom.xml#L92C2-L92C2) profile.

Modification to `spring-cloud-gcp-samples/pom.xml` so that:
- by default, all modules are included.
- To run native test only for selected modules, use `--define notAllModules=true` in mvn cmd.

This is needed because native test would fail on no test configuration found.

### To verify locally:
- Run the sample manually:
   - Follow guide in README, instead of `mvn spring-boot:run`, package with `mvn clean package -Pnative-sample-config -Pnative`
   - Run executable `./target/spring-cloud-gcp-logging-sample `
   - Follow guide to issue a request and observe log lines from cloud console.
   - Log entry written by sample:
```
{
insertId: "1c08xcxf1v07m9"
jsonPayload: {
message: "This line was also written to the log with the same Trace ID."
}
labels: {3}
logName: "projects/[project-id]/logs/spring.log"
receiveTimestamp: "2023-06-06T14:15:08.236653291Z"
resource: {2}
severity: "INFO"
timestamp: "2023-06-06T14:15:08.122Z"
trace: "projects/[project-id]/traces/trace-id-to-show-1234"
}
``` 
- Run sample test:
    - `mvn -PnativeTest clean test -Pnative-sample-config`
    - native test are run at the end of the build.
```
com.example.LoggingSampleApplicationIntegrationTests > testLogRecordedInStackDriver() SUCCESSFUL


Test run finished after 12975 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         0 tests failed          ]

```
